platform/swift/source/replay/SessionReplayTarget.swift

```diff
@@ -41,14 +42,32 @@ extension SessionReplayTarget: CapturePassable.SessionReplayTarget {
     }
 
     func captureScreenshot() {
-        // TODO: Implement
-        //        DispatchQueue.main.async { [weak self] in
-        //            self?.queue.async {
-        //                self?.logger?.logSessionReplayScreenshot(
-        //                    screen: SessionReplayCapture(data: Data()),
-        //                    duration: 0
-        //                )
-        //            }
-        //        }
+        DispatchQueue.main.async {
+            guard let window = UIApplication.shared.sessionReplayWindows().first else {
+                self.logger?.logSessionReplayScreenshot(
+                    screen: nil,
+                    duration: 0
+                )
+                return
+            }
+
+            let layer = window.layer
+            let bounds = UIScreen.main.bounds.size
+
+            self.queue.async { [weak self] in
```

Comment:
@Reflejo for my android-side understanding: would these two operations happen on separate threads? are any of these new worker threads or do we re-use existing ones?

Reply:
similarly, those these threads get spawned even when the runtime flag `session_replay.screenshots.enabled` is off?

Reply:
Queues do not map 1:1 to threads, we actually use a default queue for the [target here](https://github.com/bitdriftlabs/capture-sdk/pull/99/files#diff-99302345e23541f201ed448d77a2f32870b5c5e13ea6aa09581181a838db19f8R13) meaning this won't create an additional thread.

The lines that are under the `DispatchQueue.main.async` but below `self.queue.async` will be dispatched to the main thread (basically just to access the layer object) then the `self.queue.async` closure runs that in our serial queue mentioned before

Reply:
I'm more confused than before? then what's the diff between queue and thread? what does default mean?